e2e/testcases/otel_collector_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	monitoringv2 "cloud.google.com/go/monitoring/apiv3/v2"
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"google.golang.org/api/iterator"
+	"kpt.dev/configsync/e2e"
+	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/ntopts"
+	"kpt.dev/configsync/e2e/nomostest/retry"
+	nomostesting "kpt.dev/configsync/e2e/nomostest/testing"
+	"kpt.dev/configsync/pkg/kinds"
+	ocmetrics "kpt.dev/configsync/pkg/metrics"
+)
+
+const (
+	DefaultMonitorKSA     = "default"
+	MonitorGSA            = "e2e-test-metric-writer"
+	GCMExportErrorCaption = "failed to export time series to GCM"
+	GCMMetricPrefix       = "custom.googleapis.com/opencensus/config_sync"
+)
+
+var GCMMetricTypes = []string{ocmetrics.ReconcilerErrors.Name(), ocmetrics.PipelineError.Name(), ocmetrics.ReconcileDuration.Name(), ocmetrics.ParserDuration.Name(), ocmetrics.InternalErrors.Name()}
+
+func TestOtelCollectorDeployment(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.RequireGKE(t))
+	nt.T.Cleanup(func() {
+		nt.MustKubectl("delete", "-f", "../testdata/otel-collector/otel-cm-duplicate-timeseries.yaml", "--ignore-not-found")
+		if t.Failed() {
+			nt.PodLogs("config-management-monitoring", ocmetrics.OtelCollectorName, "", false)
+		}
+	})
+
+	// If Workload Identity enabled on cluster, setup KSA to GSA annotation
+	if workloadPool, err := getWorkloadPool(nt); err != nil {
+		nt.T.Fatal(err)
+	} else if workloadPool != "" {
+		nt.T.Log("Workload identity enabled, adding KSA annotation to use %s service account", MonitorGSA)
+		nt.MustKubectl("annotate", "serviceaccount", "--namespace", ocmetrics.MonitoringNamespace, DefaultMonitorKSA,
```

Comment:
Looks like this KSA needs a cleanup to be removed after the test. Also suggest using the golang `KubeClient` instead of shelling to kubectl when possible

Reply:
I can see the apply being done by KubeClient,  but Annotation as a new helper function to be added in the KubeClient? https://github.com/GoogleContainerTools/kpt-config-sync/blob/3dde5af1eda0b5477a28efccd63680c0646cf34a/e2e/nomostest/testkubeclient/client.go#L38

Reply:
Added annotate and deannotate function in the test using kubeClient. Do we want to do the same to the test yaml file?

Reply:
Ideally I think yes it would be preferred to inline the object with `KubeClient`, however I can see an argument for using the yaml file since it's fairly verbose

Reply:
Keep it for now?

Reply:
Yeah that's okay with me